### PR TITLE
Hd/ux quality audit fix volume alignments

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR

**This is part of the quality audit fixes.**
https://jira.unity3d.com/browse/HDRP-1035

Fix a bunch of alignment issue:
- Profile field
image

- Quality dependant field:
image

- Fix indentation only on label:
image

- Fix lesser interline when indentation decreased:
image

- Button position in HD Shadow:
image

- And last but not least, fix the alignment of field between standard inspector and volume ones:
image

To be able to better format inspector, there is two new scope available in any VolumeComponentEditor (and only this):
- IndentLevelScope (formerly internal in HDRP, now open to anyone that want to easily indent checkbox too)
- OverridablePropertyScope (handle the overridable checkbox, the disable aspect if it is uncheck, the Advanced Property and attribute decorator all in one scope)

---
### Testing status
Manual testing:
- Opened a bunch (nearly half) of all Volume (mix HDRP anr URP), display is ok
- Tested attribute work fine with both the scope and the automatic PropertyField (Tooltip, InspectorName, Header, Space, AdditionalProperty), In different scenario (default inspector, default PropertyField drawer, custom drawer with OverridablePropertyScope) 
**NOT TESTED:** parameter with custom drawer (without and without custom override checkbox drawer. Current usage works in tested volume component)
- Ensuring inspector part for ObjectParameter work again
image

Script used for test:
[VolumeComponentTest.cs.txt](https://github.com/Unity-Technologies/Graphics/files/6637196/VolumeComponentTest.cs.txt) (remove .txt)

---

### CODE REVIEW

**Review by JulienIgnace-Unity — APPROVED** [review]

File: com.unity.render-pipelines.core/CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed a large, visible stretch ratio in a LensFlare Image thumbnail.
 - Fixed Undo from script refreshing thumbnail.
 - Fixed cropped thumbnail for Image with non-uniform scale and rotation
+- Fixed alignment in Volume Components
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.core/CHANGELOG.md:36]:
```suggestion
- Fixed alignment in Volume Components.
```

File: com.unity.render-pipelines.core/Editor/Volume/VolumeComponentEditor.cs [hunk 2]
```diff
@@ -626,13 +609,149 @@ protected void DrawOverrideCheckbox(SerializedDataParameter property)
...
+                    editor.EndAdditionalPropertiesScope();
+
+                EditorGUIUtility.labelWidth += Styles.overrideCheckboxWidth + Styles.overrideCheckboxOffset + 3 + 3;
+            }
+        }
+
+        /// <summary>
+        /// Like EditorGUI.IndentLevelScope but this one will also indent the override checkboxes.
+        /// </summary>
+        protected class IndentLevelScope : GUI.Scope
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.core/Editor/Volume/VolumeComponentEditor.cs:728]: For me is not clear that this scope is indenting the overrides too, can we find a better name?

**Review by alex-vazquez — APPROVED** [review]

**Review by TheoWong-pixel — APPROVED** [review]

**Review by simon-engelbrecht-soerensen — APPROVED** [review]
URP changes 👍

**Review by JMargevics — APPROVED** [review]
Checked on HDRP side. No inspector functionality broken. Everything looks aligned and tidy ✅

**Comment by sebastienlagarde** [comment]
URP side is a single file with very minimal change. Merging

**Pull request merged**